drivers/serial/Kconfig.ns16550

```diff
@@ -25,6 +25,16 @@ config UART_NS16550_DLF
 
 	  Says n if you are not sure if hardware supports this.
 
+config UART_NS16550_PCP
+	bool "Enable Apollo Lake PRV_CLOCK_PARAMS (PCP) support"
```

Comment:
is this really an exclusive feature of apollo lake or is that a feature of the NS116500 used on that platform. Lets try and keep the drivers agnostic to the platform they are being used on.

Reply:
As far as I know, this is specific to the Apollo Lake's implementation of the NS16550.  Unfortunately this configuration isn't board- or SoC-level; rather each UART has its own clock block which is set by manipulating device registers, which requires knowledge of where and how the device is mapped, and so the driver is IMO "best" place to put this.

I agree that this pollution of the driver is suboptimal, but it does follow the example set for the DLF register, which is specific to the NS16550 implementation found on some of the Quarks.

Reply:
DLF is a Synopsys addition to the IP block which applies to more than one SoC. For this, I wonder if we should set up the clock in early SoC init code.